src/main/java/com/elastic/support/config/ElasticClientInputs.java

```diff
@@ -0,0 +1,145 @@
+    protected boolean skipVerification = false;
+    @Parameter(names = {"--keystore"}, description = "Keystore for client certificate.")
+    protected String pkiKeystore;
+    @Parameter(names = {"--keystorePass"}, description = "Keystore password for client certificate.", password = true)
+    protected String pkiKeystorePass;
+
+    public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    public String getUser() {
+        return user;
+    }
+
+    public void setUser(String username) {
+        this.user = user;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public boolean isSsl() {
+        return isSsl;
+    }
+
+
+    public String getScheme() {
+        if (this.isSsl) {
+            return "https";
+        } else {
+            return "http";
+        }
+    }
+
+    public String getPlainTextPassword() {
+        return plainTextPassword;
+    }
+
+    public void setPlainTextPassword(String plainTextPassword) {
+        this.plainTextPassword = plainTextPassword;
+    }
+
+    public boolean isSkipVerification() {
+        return skipVerification;
+    }
+
+    public void setSkipVerification(boolean skipVerification) {
+        this.skipVerification = skipVerification;
+    }
+
+    public String getPkiKeystore() {
+        return pkiKeystore;
+    }
+
+    public void setPkiKeystore(String pkiKeystore) {
+        this.pkiKeystore = pkiKeystore;
+    }
+
+    public String getPkiKeystorePass() {
+        return pkiKeystorePass;
+    }
+
+    public void setPkiKeystorePass(String pkiKeystorePass) {
+        this.pkiKeystorePass = pkiKeystorePass;
+    }
+
+    public boolean isSecured() {
+        return (StringUtils.isNotEmpty(this.user) && StringUtils.isNotEmpty(this.password));
+    }
+
+    public boolean isPki(){
+        if(StringUtils.isEmpty(pkiKeystore) ){
+            return false;
+        }
+        return true;
+    }
+
+    public boolean validate() {
+
+        if( ! super.validate() ){
+            return false;
+        }
+
+        if (StringUtils.isNotEmpty(this.plainTextPassword)) {
+            this.password = plainTextPassword;
+        }
+
+        if (!isAuthValid(user, password)) {
+            logger.info("Input error: If authenticating both user and password are required.");
+            this.jCommander.usage();
+            return false;
+        }
+
+        return true;
+
+    }
+
+    protected boolean isAuthValid(String user, String password) {
+
+        return (!StringUtils.isNotEmpty(user) || !StringUtils.isEmpty(password)) &&
```

Comment:
I had to re-read this a few times because of all of the `!` inversions happening (compounded by some of the methods themselves implicitly including `Not`). This may be a little easier to follow?

```java
boolean invalid =
  (StringUtils.isEmpty(user) && StringUtils.isNotEmpty(password)) ||
  (StringUtils.isNotEmpty(user) && StringUtils.isEmpty(password));

return !invalid;
```